ops/charm.py

```diff
@@ -297,86 +245,71 @@ def restore(self, snapshot):
             self.unit = None
 
 
+class RelationCreatedEvent(RelationEvent):
+    """Represents the `relation-created` hook from Juju.
+
+    A relation-created event fires after a relation is created in Juju. A relation-joined
+    event will only fire after the respective relation-created event.
+    For units that are coming up, relation-created events fire after the install event
+    before any other events that fire after install, including leader-elected. This allows
+    a leader unit to work with peer application relation data early in its life cycle.
```

Comment:
We should also confirm that this reads clearly, I think it was the last bit that Gustavo was concerned about.

Reply:
It isn't clear in and of itself. Perhaps reiterating the order of events would help, as that is where it gets confusing. Or we could drop the whole thing after the first sentence for now and add the ordering elsewhere; it does feel like there needs to be a centralised "this is the order of things" somewhere, not spread out amongst the events.

Reply:
Shouldn't this be explicited in some *Juju* docs? And we can just point to that.